pkg/client/client.go

```diff
@@ -59,8 +58,9 @@ func NewHTTPClient(opts Options) *HTTPClient {
 		ExpectContinueTimeout: 1 * time.Second,
 		DisableKeepAlives:     disableKeepAlives,
 	}
-	transport.MaxConnsPerHost = opts.MaxConnPerHost
-	transport.MaxIdleConnsPerHost = opts.MaxConnPerHost
+
+	// FIXME should we set this to something larger than the default of 2?
+	// transport.MaxIdleConnsPerHost = opts.MaxConnPerHost
```

Comment:
Let's make sure to update this before we land the code. Easiest is to set it == concurrency. We could just do Concurrency+5 for super easy math or some such.